src/main/java/appeng/client/texture/TextureUtils.java

```diff
@@ -0,0 +1,55 @@
+package appeng.client.texture;
+
+import javax.annotation.Nonnull;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.client.renderer.texture.TextureMap;
+import net.minecraft.util.IIcon;
+
+import cpw.mods.fml.relauncher.Side;
+import cpw.mods.fml.relauncher.SideOnly;
+
+public class TextureUtils {
+
+    @SideOnly(Side.CLIENT)
+    @Nonnull
+    public static IIcon checkTexture(IIcon val) {
+        return checkTexture(val, true);
+    }
+
+    @SideOnly(Side.CLIENT)
+    @Nonnull
+    public static IIcon checkTexture(IIcon val, boolean isBlock) {
+        if (val == null) {
+            val = isBlock ? getMissingBlock() : getMissingItem();
+        }
+
+        assert val != null;
+
+        return val;
+    }
+
+    private static TextureAtlasSprite missingBlockTexture = null;
```

Comment:
The fields should probably also be SideOnly

Reply:
Can I just apply SideOnly to the entire class?

Reply:
Yes, but it does mean that if any common code refers to the class at all it will crash. I'd rather the class exist and be empty on the server side if loaded by accident than crash in some random place.

Reply:
Gotcha, that makes sense.